libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFNetwork.m

```diff
@@ -41,6 +45,42 @@ @implementation SFNetwork
 static NSURLSessionConfiguration *kSFSessionConfig;
 __weak static id<SFNetworkSessionManaging> kSFNetworkManager;
 
+- (instancetype)initWithSessionConfigurationIdentifier:(nonnull NSString *)identifier sessionConfiguration:(nullable NSURLSessionConfiguration *)sessionConfiguration useSharedSession:(BOOL)useSharedSession {
+    self = [super init];
+    if (self) {
+        if (kSFNetworkManager) {
+            self.activeSession = [kSFNetworkManager sessionWithConfigurationIdentifier:identifier sessionConfiguration:sessionConfiguration useSharedSession:useSharedSession];
+            return self;
+        }
+
+        NSURLSession *session;
+        if (useSharedSession) {
+            session = [[self class] sharedSessions][identifier];
+        }
+
+        if (!session) {
+            if (!sessionConfiguration) {
+                if ([identifier isEqualToString:kSFNetworkBackgroundSessionIdentifier]) {
+                    sessionConfiguration = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:identifier];
+                } else {
+                    // Case 1: Identifier is `kSFNetworkEphemeralSessionIdentifier`
+                    // Case 2: Custom identifier provided but session configuration wasn't, default to ephemeral as well
+                    sessionConfiguration = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+                }
+            }
+
+            if (useSharedSession) {
+                session = [NSURLSession sessionWithConfiguration:sessionConfiguration delegate:self delegateQueue:nil];
```

Comment:
Why is the delegate set here but not in the else case?

Reply:
The delegate method `(void)URLSession:(NSURLSession *)session didBecomeInvalidWithError:(nullable NSError *)error` is only used to remove items from the shared session dictionary. I can change it so that the delegate it used for both cases and have a check in the method instead if that's preferable (that'd also be good if we want to do anything with the error)

Reply:
I see. If you set the delegate for the non-shared case, how would you know that the NSURLSession is shared or not? Is it possible to create two NSURLSession with the same identifier, one shared and one not shared?